File: algorithms/1_easy/C/01598_crawler_log_folder/solution.cpp
class Solution {
public:
    int minOperations(vector<string>& logs) {
        int curr{ 0 };

        for (const auto& log : logs) {
            if (log == "../") {
                curr = curr > 0 ? curr - 1 : 0;
            } else if (log == "./") {
                continue;
            } else {
                curr += 1;
            }
        }

        return curr;
    }
};

File: algorithms/1_easy/C/01598_crawler_log_folder/solution.rs
impl Solution {
    pub fn min_operations(logs: Vec<String>) -> i32 {
        let mut curr = 0_i32;

        for log in logs.into_iter() {
            match log.as_str() {
                "../" => curr = if curr == 0 { 0 } else { curr - 1 },
                "./" => {},
                _ => curr += 1,
            }
        }

        curr
    }
}
